### ISSUE TITLE
Why not upgrade express from version 3.x to 4.x?

### ISSUE THREAD

**Issue by huguangju** [balderdashy/sails#2938]
[Migrating from 3.x to 4.x](https://github.com/strongloop/express/wiki/Migrating-from-3.x-to-4.x)

**Comment by tjwebb** [comment]
This is something we'd love to do in the next release. If you'd like to help, a PR would be appreciated. 

cc @loicsaintroch

**Comment by listepo** [comment]
I think it would be logical to make these changes after https://github.com/balderdashy/sails/issues/2833 to see how the changed code coverage.

**Comment by listepo** [comment]
I started to work. I would be grateful for the help. https://github.com/listepo/sails/tree/next

**Comment by nikhiljohn10** [comment]
Currently A default sails app is eating 140 - 160 MB of my server RAM and it increase as the app is modified. I create a Express 4.0 App. It used only 20 - 70 MB. Can anyone explain why the RAM usage is high is Sails?

**Comment by tjwebb** [comment]
"eating 140MB of RAM" is not a bug. There are *tons* of other memory usage threads in github that you can search. Please read through those before you file memory usage issues.

**Comment by nikhiljohn10** [comment]
Ok.

**Comment by sailsbot** [comment]
Thanks for posting, @huguangju.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by mikermcneil** [comment]
> Currently A default sails app is eating 140 - 160 MB of my server RAM and it increase as the app is modified. I create a Express 4.0 App. It used only 20 - 70 MB. Can anyone explain why the RAM usage is high is Sails?

@nikhiljohn10 The rest of the core team and I spent a lot of time analyzing this and other memory concerns earlier this year when we were first scaling the production deployment of Treeline.  What we learned is that the 140-160MB RAM usage in development is probably due to the Grunt child process running in the background, along with the fact that Sails has more dependencies than Express.  We've never experienced any sort of problems due to this alone (although there have been memory leak issues stemming from the release of Node v0.12, but those have since been resolved in a patch release to Node core-- be sure and check your versions though).

Hope that helps!

**Comment by mikermcneil** [comment]
@huguangju re: upgrading Express, see https://github.com/balderdashy/sails/issues/3460#issuecomment-169545356 for our plans.

**Comment by mikermcneil** [comment]
(locking this issue to funnel future discussion into the linked PR so it's easier to track)

**Comment by mikermcneil** [comment]
For future reference, there's more information and tips on diagnosing memory leaks in Node apps, and on reproducing+reporting suspected memory leaks in Sails here:
https://github.com/balderdashy/sails/issues/3782#issuecomment-238679052
